reconcile/queries.py

```diff
@@ -2754,3 +2760,24 @@ def get_blackbox_exporter_monitoring_provider() -> dict:
 def get_jenkins_configs():
     gqlapi = gql.get_api()
     return gqlapi.query(JENKINS_CONFIGS)["jenkins_configs"]
+
+
+TF_RESOURCES_PROVIDER_EXCLUSIONS_BY_PROVISIONER = """
+{
+  tf_provider_exclusions_by_provisioner: app_interface_settings_v1 {
+    terraformResourcesProviderExclusionsByProvisioner {
+      provisioner {
+          name
+      }
+      excludedProviders
+    }
+  }
+}
+"""
+
+
+def get_tf_resources_provider_exclusions_by_provisioner():
+    gqlapi = gql.get_api()
+    return gqlapi.query(TF_RESOURCES_PROVIDER_EXCLUSIONS_BY_PROVISIONER)[
+        "tf_provider_exclusions_by_provisioner"
+    ]
```

Comment:
can we move this to typed queries?

Reply:
In this specific case, it's better to stick to queries. This attribute normally comes from app_interface_settings which is obtained using `queries`. This code is only used if the attribute is None to ensure the exclusions are taken into account.  

With a typed query, the code to build the dictionary from your other suggestion will need to handle both cases nad I think it's not necessary in this specific case.